### ISSUE TITLE
Sequelize CLI not always detecting migration changes?

### ISSUE THREAD

**Issue by tylerFowler** [sequelize/cli#140]
Perhaps I'm just doing/understanding it wrong, but I notice that when I modify my migrations the changes aren't detected when trying to run migrations? For example, I have a few tables (one of which is populated, I'm still early in development & piloting), and one (that is empty as I'm working on the feature) I removed a `UNIQUE` constraint. Does the CLI currently detect changes like this? Or is it just checking for the presence of the table?

Obviously it's not too much trouble to reflect this change manually but it would be nice to be able to do smaller changes like this.

**Comment by redhotvengeance** [comment]
Migrations flow in one direction at a time: up or down. Running migrations doesn't iterate through all of the previously run migrations to look for differences - it just checks to see if there are any new migrations to be run.

So you have two options in your situation. If you haven't shared the new migrations with any teammates and haven't deployed the migrations, then you can rollback your local database with `sequelize db:migrate:undo`, change the migration (remove the `UNIQUE` constraint), and then migrate again (`sequelize db:migrate`).

If you have shared the migrations and/or deployed already, you don't want to alter the migration history (otherwise it could collide when rolling back). So you'll want to add a new migration that makes the desired change (in your case, removing the `UNIQUE` constraint).

**Comment by tylerFowler** [comment]
I see. Well in this case, since we're rolling out this particular project in phases (as the requirements change) is there currently a way to do this without dropping all the tables? Selectively undoing/migrating models I mean.

**Comment by redhotvengeance** [comment]
Sure, just add a new migration to remove the `UNIQUE` constraint. Then all you'll be doing is migrating forward, and not rolling back anything.

**Comment by tylerFowler** [comment]
Interesting, I thought that that might screw up the data the CLI uses in the meta table. Well that will probably suit my needs just fine, thanks!

**Comment by sdepold** [comment]
Thanks for answering the ticket :) In general the idea of migrations is to have every change on the database in it's own migration, especially when things have been rolled out already. In reality you will most likely find situations like yours where you are developing something and are potentially not interested in your current data. In that cases you can easily just reset your db (as in `sequelize db:migrate:undo:all`) and migrate all the things again.

**Comment by RusinovAnton** [comment]
> In that cases you can easily just reset your db (as in sequelize db:migrate:undo:all) and migrate all the things again.

I just did exactly the same and still no effect. I'm trying to add new column to my table, but no luck.
What I did is generated a model with `model:generate` command, then when I needed new column I did the same, adding new column in `--attributes` option also adding `--force` option to rewrite model and migration, then dropped database and created, migrated and seeded it again, but my new column won't appear. I also used `migrate:undo:all`. What's the magic?

**Comment by papb** [comment]
@RusinovAnton Can you please post the contents of your SequelizeMeta table and the file names of all your migrations? And which one is the one that is not running? Thanks

**Comment by RusinovAnton** [comment]
Hey, @papb. Thanks for having time to take a look at it. 

**TL;DR I've deleted old migration file to make a new one work**

In a nutshell, I've just created a new project to reproduce an issue and here are my migration files:

```
20191010182858-create-example-model.js
20191010183123-create-example-model.js
```

I have them in SequelizeMeta like this:
```
name
-----
20191010182858-create-example-model.js
20191010183123-create-example-model.js
```

and these are my dependencies:
```
"sequelize-cli": "^5.5.1"
"pg": "^7.12.1",
"sequelize": "^5.19.5"
```

Here is repo with reproducible example and step-by-step explanation of my commands:
https://github.com/RusinovAnton/sequelize-model-example

Hope that helps

**Comment by papb** [comment]
@RusinovAnton Awesome, thanks! My time is super short though :( I won't be able to look into it today. Hopefully I'll have time in the near future...
